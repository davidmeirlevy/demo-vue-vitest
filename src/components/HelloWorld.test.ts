import {mount} from '@vue/test-utils'
import {nextTick} from 'vue';

describe('HelloWorld', () => {

	const {'./HelloWorld.vue': {default: HelloWorld}} = import.meta.globEager('./HelloWorld.vue');

	test('should render', () => {
		const wrapper = mount(HelloWorld, {props: {msg: 'dasdsad'}});

		expect(wrapper.findComponent('HelloWorld')).toBeDefined();
	});

	test('should render msg', () => {
		const wrapper = mount(HelloWorld, {props: {msg: 'my test'}});

		expect(wrapper.find('h1').text()).toBe('my test');
	});

	test('should show zero in counter', () => {
		const wrapper = mount(HelloWorld, {props: {msg: 'my test'}});

		expect(wrapper.find('button').text()).toBe('count is: 0');
	});

	test('should show 1 in counter', async () => {
		const wrapper = mount(HelloWorld, {props: {msg: 'my test'}});

		expect(wrapper.find('button').text()).toBe('count is: 0');

		await wrapper.find('button').trigger('click');

		await nextTick();

		expect(wrapper.find('button').text()).toBe('count is: 1');

		expect(wrapper.html()).toMatchSnapshot();

	});
})
