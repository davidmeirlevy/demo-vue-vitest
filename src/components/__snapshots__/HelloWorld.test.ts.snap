// Vitest Snapshot v1

exports[`HelloWorld > should show 1 in counter 1`] = `
"<h1 data-v-469af010=\\"\\">my test</h1>
<button type=\\"button\\" data-v-469af010=\\"\\">count is: 1</button>"
`;
